static/js/hotkey.js

```diff
@@ -571,6 +571,52 @@ exports.process_hotkey = function (e, hotkey) {
         return false;
     }
 
+    if (event_name === "next_unread_topic") {
+        // message_header of selected_message
+        var message_header = $(".selected_message").closest(".recipient_row").find(".message_header");
+        var selected_stream = ($(message_header).find(".stream_label").html()).replace(/ |\n/g,'');
+        var selected_topic = ($(message_header).find(".stream_topic a").text()).replace(/ |\n/g,'');
+        // console.log(selected_stream);
+        // console.log(selected_topic);
+        // Returns the next unread topic.
+        var stream_unread_name;
+        var topic_unread_name;
+        var select_this_topic = 0;
+        var stream_unread = unread.get_counts().subject_count._items;
+        function next_topic() {
+            for (var key in stream_unread) {
+                var stream_unread_dict = stream_unread[key];
+                stream_unread_name = stream_unread[key].k;
+                if (stream_unread_name === selected_stream || select_this_topic) {
+                    var topic_unread = stream_unread_dict.v._items;
+                    for (var key1 in topic_unread) {
+                        topic_unread_name = topic_unread[key1].k;
+                        // console.log(topic_unread_name, select_this_topic);
+                        if (topic_unread_name === selected_topic || select_this_topic) {
+                            if (select_this_topic) {
+                                return;
+                            }
+                            select_this_topic = 1;
+                        }
+                    }
+                }
+            }
+            // Incase the selected topic is the last one in the list,
+            // Call the function again and return first unread topic.
+            select_this_topic = 1;
+            next_topic();
+        }
+        next_topic();
+        // console.log(stream_unread_name, topic_unread_name);
+        // go to the topic
+        var search_terms = [
+            {operator: 'stream', operand: stream_unread_name},
+            {operator: 'topic', operand: topic_unread_name},
+        ];
+        narrow.activate(search_terms, {select_first_unread: true, trigger: 'hotkeys'});
+        return true;
+    }
```

Comment:
Move most of this code to the `unread_ops` module.  that will help with testing it.

Also, change `next_topic` to return values (and thus be a testable function) rather than modifying variables in its caller.  We'll want to write an extensive set of unit tests for `next_topic`  that set up some state and then verify that the right output comes out, similar to e.g. `frontend_tests/node_tests/unread.js`.